spec/support/helpers/temporary_table_helper.rb

```diff
@@ -0,0 +1,52 @@
+module TemporaryTableHelper
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def with_temporary_table(name, columns = ->(t){}, create_table: true, &block)
+      table_builder = nil
+
+      before(:context) do
+        ActiveRecord::Base.connection.schema_cache.clear!
+        ActiveSupport::Dependencies::Reference.clear!
+        table_builder = TableBuilder.new(name, columns, create_table: create_table, &block)
+      end
+
+      after(:context) do
+        table_builder.teardown
+      end
+    end
+  end
+
+  class TableBuilder
+    def initialize(name, columns, create_table: true, &block)
+      model_name = name.to_s.classify
+
+      if create_table
+        table_name = name.to_s.pluralize
+        ActiveRecord::Base.connection.create_table(table_name, &columns)
+      end
+
+      @model_class =
+        Class.new(ActiveRecord::Base) do |c|
+        raise "Model already defined: #{model_name}" if Object.const_defined?(model_name, false)
+        Object.const_set(model_name, self)
+        self.primary_key = :id
+        class_eval(&block) if block_given?
+        reset_column_information
+      end
+    end
+
+    def teardown
+      if @model_class.nil?
+        raise ArgumentError.new("Can't destroy test model #{name.inspect} - missing!")
+      end
+
+      if @model_class.table_exists?
+        table_name = @model_class.table_name
+        @model_class.connection.drop_table(table_name)
+      end
+
+      Object.send(:remove_const, @model_class.name)
```

Comment:
Even though it's only used in this line? Just extract the expression `@model_class.name` to a variable `model_name`?